opt_einsum/contract.py

```diff
@@ -397,7 +419,7 @@ def _core_contract(operands, contraction_list, **einsum_kwargs):
         handle_out = specified_out and ((num + 1) == len(contraction_list))
 
         # Call tensordot
-        if blas:
+        if blas or not has_einsum:
```

Comment:
Not all operations can be written as BLAS, what happens if a backend does not have einsum?

Reply:
Yeah this is a bit unsatisfactory, my suggestion is this, have ``can_blas`` return ``False`` for 'can't blas' and ``None`` for 'prefer einsum' (i.e. blas still possible), then this condition will look like ``if blas or (blas is None and not has_einsum)``. Then any times where a backend doesn't work will raise an ``AttributeError`` saying the backend doesn't have einsum, which is preferable.